src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,345 @@
+            'size' => $sizeComponent
+        ));
+    }
+
+    //.......Share Dashboard Function
+    /***
+     * @param Request $request
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function shareAction(Request $request, $id)
+    {
+        //.......Get current dashboard
+        $em = $this->getDoctrine()->getManager();
+        $dashboard = $em
+            ->getRepository('AppBundle:Dashboard')
+            ->find($id);
+        //.......Get the parameter from url for search function
+        $parameter = $request->get('search');
+        //.......Get all the user that have in the last name the "value" parameter
+        $listUsers = $em
+            ->getRepository('AppBundle:User')
+            ->createQueryBuilder('user')
+            ->where('user.lastName LIKE :lastName')
+            ->setParameter('lastName', '%' . $parameter . '%')
+            ->getQuery()
+            ->getResult();
+
+        //.......if form has been submitted removes all collaborators from dashboard,
+        // ......adds the checked ones to the dashboard and saves them in the database
+        if ($request->get('activeUsers')) {
+            $activeUsers = $request->get('activeUsers');
+
+            foreach ($listUsers as $inactiveUser) {
+                $dashboard->removeCollaborator($inactiveUser);
+                $em->persist($dashboard);
+            }
+
+            foreach ($activeUsers as $active) {
+                $user = $em->getRepository('AppBundle:User')
+                    ->find($active);
+                $dashboard->addCollaborator($user);
+                $em->persist($dashboard);
+            }
+
+            $em->flush();
+
+            return $this->redirectToRoute('app_dashboard', array(
+                'id' => $id));
+        }
+        //......pagination
+        $users = $this->get('knp_paginator')->paginate($listUsers,
+            $this->get('request')->query->get('page', 1), 4/*limit per page*/
+        );
+
+        return $this->render('AppBundle:App:shareD.html.twig', array(
+            'listUsers' => $listUsers,
+            'users' => $users,
+            'dashboard' => $dashboard
+        ));
+    }
+
+    //.......Delete dashboard
+    /***
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function deleteDashAction($id)
+    {
+        //.......Get current dashboard
+        $em = $this->getDoctrine()->getManager();
+        $dashboard = $em->getRepository('AppBundle:Dashboard')->find($id);
+        //.......get the components that are on the dashboard
+        $components = $em->getRepository('AppBundle:Component')->findBy(
+            array('dashboard' => $id));
+
+        //.......delete the components
+        //.......delete the dashboard and update database
+        if ($dashboard != null) {
+            foreach ($components as $component) {
```

Comment:
If the dashboard knows its components, then you can configure it to cascade the delete operation on the components too.

It would look like

```
/**
 * @ORM\OneToMany(targetEntity="AppBundle\Entity\Component", cascade={"remove"}
 */
$components;
```

and then you just have to write 
```
$em->remove($dashboard);
$em->flush();
```